app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class MicropostsController < ApplicationController
+  before_action :restrict, only: %i[new create]
+
+  def new
+    @micropost = Micropost.new
+  end
+
+  def create
+    @micropost = Micropost.new(user_params)
+    @micropost.user = current_user
```

Comment:
These lines could be refactored using the `build` method from the association between users and microposts.